my/src/views/Apply.vue

```diff
@@ -206,42 +207,36 @@ export default Vue.extend({
     },
     onFormChange() {
       if (this.updateTimeout) clearTimeout(this.updateTimeout);
-      this.updateTimeout = setTimeout(() => {
-        this.snack.message = 'Progress saved!';
-        this.snack.color = 'success';
-        this.updateAppProgress(false);
-      }, 4000);
+      this.updateTimeout = setTimeout(() => this.updateAppProgress(false), 4000);
     },
 
     async updateAppProgress(submitting: boolean) {
-      let submit = false;
-      const verified = await (firebase.auth().currentUser as firebase.User)
-        .emailVerified;
-      if (submitting && this.app._.status === 'in progress' && verified) {
-        this.app._.status = 'submitted';
-        this.app._.time_submitted = new Date();
-        this.snack.message = 'Application submitted';
+      let updateError;
+
+      try {
+        const updateResponse = await firebaseMaster
+          .functions()
+          .httpsCallable('updateApplication')({
+            app: this.app,
+            isSubmission: submitting,
+          });
+        console.log(updateResponse)
+        if (updateResponse.data.error) {
+          updateError = updateResponse.data.error;
+        } else {
+          this.snack.message = submitting ? 'Application submitted' : 'Progress saved!';
         this.snack.color = 'success';
-        submit = true;
-      } else if (
-        submitting &&
-        this.app._.status === 'in progress' &&
-        !verified
-      ) {
-        this.snack.message = 'Please verify your email before submitting!';
```

Comment:
We're verifying emails on the back-end instead 🎉

Reply:
Oh, and the '_' property is basically read-only for the front-end, so there's a bunch of dead code we'll have to get rid of. I'll do that later to keep this PR small